### PR TITLE
Rolling upgrade 168948431

### PR DESCRIPTION
**Pull request by ablease**
Second draft of our rolling upgrade document. 

https://www.pivotaltracker.com/story/show/168948431

### CODE REVIEW

File: rolling-upgrade.html.md.erb [hunk 1]
```diff
@@ -4,27 +4,52 @@ owner: London Services
 ---
 
 ## What is a rolling upgrade
-A rolling upgrade is a rollout strategy for updating software in a distributed system. Rolling upgrades guarantee at least one node will remain up throughout the upgrade process for availabilty at all times. Before 1.17.3, some scenarios required the whole cluster to be shutdown in order to perform an upgrade. These scenarios included where the major or minor version of the underlying RabbitMQ software was updated, or where the major version of the underlying Erlang distribution was updated.
+A rolling upgrade is a rollout strategy for updating software in a distributed
+system. Rolling upgrades guarantee at least one node will remain up throughout
+the upgrade process for availability at all times. Before 1.17.3 some upgrades
```
Comment (edbyford) [inline rolling-upgrade.html.md.erb]: The VM will remain up - and this should = the node, but it's not always the case. I therefore don't think we can guarantee this.

File: rolling-upgrade.html.md.erb [hunk 2]
```diff
@@ -4,27 +4,52 @@ owner: London Services
...
+required the whole cluster to be shut down. For example when the major or minor
```
Comment (edbyford) [inline rolling-upgrade.html.md.erb]: For readability, this should be a semi-colon, not a full stop

File: rolling-upgrade.html.md.erb [hunk 3]
```diff
@@ -4,27 +4,52 @@ owner: London Services
...
+version of RabbitMQ was updated, or when the major version of the Erlang
+distribution was updated.
+
+As of 1.17.3, upgrades are now performed using a rolling upgrade strategy.
+Today, the only case where a cluster is required to fully shut down as part of
```
Comment (edbyford) [inline rolling-upgrade.html.md.erb]: Remove today

File: rolling-upgrade.html.md.erb [hunk 4]
```diff
@@ -4,27 +4,52 @@ owner: London Services
...
+an upgrade is where the Erlang cookie for the cluster is changed. However as
```
Comment (edbyford) [inline rolling-upgrade.html.md.erb]: Remove "However"

File: rolling-upgrade.html.md.erb [hunk 5]
```diff
@@ -67,4 +106,22 @@ applications:
...
\ No newline at end of file
+We successfully observed that the downtime experienced during this rolling
+upgrade was significantly reduced compared to a similar upgrade where the
+cluster is brought fully down. Our metrics indicated that the 'downtime', i.e.
+where a publisher was unable to publish a message to a queue, was at maximum 5
+seconds. This is due to the fact that the internal BOSH DNS record used to
+round-robin messages to the nodes in the cluster has a TTL of 5 seconds. The
+messages may be routed to nodes which have just been replaced. As our app has
+retry logic we did not observe service outage.
```
Comment (edbyford) [inline rolling-upgrade.html.md.erb]: We should add a link to Marcial's resiliency repo here

**Comment by RichardJJG** [comment]
Thanks, @ablease.

**Pull request merged**